### PR TITLE
Fix few compiler warnings. While here also add NULL checker to sendfile.

### PR DESCRIPTION
**Pull request by ghost**
I'd like to get rid of this spam of warnings:
```
In file included from /usr/local/pobj/warsow-2.1.2/warsow-2.1.2-qf/source/ref_gl/r_glimp.h:68:
/usr/local/pobj/warsow-2.1.2/warsow-2.1.2-qf/source/ref_gl/qgl.h:553:43: warning: extra ';' outside of a function [-Wextra-semi]
QGL_WGL(PROC, wglGetProcAddress, (LPCSTR));
                                          ^
/usr/local/pobj/warsow-2.1.2/warsow-2.1.2-qf/source/ref_gl/qgl.h:554:73: warning: extra ';' outside of a function [-Wextra-semi]
QGL_WGL(int, wglChoosePixelFormat, (HDC, CONST PIXELFORMATDESCRIPTOR *));
                                                                        ^
/usr/local/pobj/warsow-2.1.2/warsow-2.1.2-qf/source/ref_gl/qgl.h:555:80: warning: extra ';' outside of a function [-Wextra-semi]
QGL_WGL(int, wglDescribePixelFormat, (HDC, int, UINT, LPPIXELFORMATDESCRIPTOR));
                                                                               ^
/usr/local/pobj/warsow-2.1.2/warsow-2.1.2-qf/source/ref_gl/qgl.h:556:76: warning: extra ';' outside of a function [-Wextra-semi]
QGL_WGL(BOOL, wglSetPixelFormat, (HDC, int, CONST PIXELFORMATDESCRIPTOR *));

...
```

Should I just remove extra semicolons or is there a way to fix defines in r_glimp.h?

### CODE REVIEW

**Comment by ghost** [comment]
@viciious this is undefined behavior:
```
/usr/local/pobj/warsow-2.1.2/warsow-2.1.2-qf/source/unix/unix_fs.c:499:14: warning: arithmetic on a pointer to void is a GNU extension [-Wpointer-arith]
        return data + offsetpad;
               ~~~~ ^
```

**Comment by ghost** [comment]
@viciious what's the meaning of this?
```
[  5%] Building C object ref_gl/CMakeFiles/ref_gl.dir/r_backend_program.c.o
/home/travis/build/Qfusion/qfusion/source/cin/cin_theora.c:429:22: warning: 
      ordered comparison between pointer and zero
      ('cin_raw_samples_listener_t *' and 'int') is an extension [-Wpedantic]
                if( cin->listeners > 0 ) {
                    ~~~~~~~~~~~~~~ ^ ~
```

**Comment by ghost** [comment]
@viciious wtf, that comparison makes no sense at all. That pointer is a fixed array to a structure with pointers. What was originally intended?

**Comment by ghost** [comment]
@viciious Ok, never mind that, I see you've just made a misspell.

File: source/CMakeLists.txt [hunk 1]
```diff
@@ -121,7 +121,7 @@ elseif (APPLE)
     set(CMAKE_OSX_ARCHITECTURES "$(ARCHS_STANDARD_64_BIT)")        # Compile only 64-bit version
     set(CMAKE_OSX_SYSROOT "macosx")                                # Compile with latest available OS X sdk
     set(CMAKE_XCODE_ATTRIBUTE_MACOSX_DEPLOYMENT_TARGET "10.8")     # Use 10.8 as deployment target
-    set(CMAKE_XCODE_ATTRIBUTE_GCC_C_LANGUAGE_STANDARD "gnu99")     # Use GNU99 standart for compiling C files
+    set(CMAKE_XCODE_ATTRIBUTE_GCC_C_LANGUAGE_STANDARD "gnu11")     # Use GNU99 standart for compiling C files
```
Comment (viciious) [inline source/CMakeLists.txt]: The comment is wrong

**Review by viciious — COMMENTED** [review]
Why didn't you fix the defines in r_glimp.h instead?

**Review by viciious — COMMENTED** [review]
Why? Without this sleep call, the loop becomes a hot one and busy-waits for the next frame.

**Comment by ghost** [comment]
@viciious comment is fixed, sleep call removal is reverted. I see no way to fix defines in r_glimp.h, if they were used in some function I'd just use "do { ... } while (0)", but they are not.

**Review by viciious — COMMENTED** [review]
Please do not remove sdl_threads - it's there in case the OS-specific code gets eventually removed.

**Comment by viciious** [comment]
> @viciious comment is fixed, sleep call removal is reverted. I see no way to fix defines in r_glimp.h, if they were used in some function I'd just use "do { ... } while (0)", but they are not.

Why would removal of the semicolons from the glimp.h macros not work?

**Comment by ghost** [comment]
@viciious because there are empty defines, when they get expanded we end up having extra semicolons. For example, on systems other than Windows this:
```
QGL_WGL( PROC, wglGetProcAddress, ( LPCSTR ) );
QGL_WGL( int, wglChoosePixelFormat, ( HDC, CONST PIXELFORMATDESCRIPTOR * ) );
QGL_WGL( int, wglDescribePixelFormat, ( HDC, int, UINT, LPPIXELFORMATDESCRIPTOR ) );
QGL_WGL( BOOL, wglSetPixelFormat, ( HDC, int, CONST PIXELFORMATDESCRIPTOR * ) );
QGL_WGL( BOOL, wglSwapBuffers, ( HDC ) );
QGL_WGL( HGLRC, wglCreateContext, ( HDC ) );
QGL_WGL( BOOL, wglDeleteContext, ( HGLRC ) );
QGL_WGL( BOOL, wglMakeCurrent, ( HDC, HGLRC ) );
QGL_WGL( BOOL, wglShareLists, ( HGLRC, HGLRC ) );
```
Gets expanded to:
```
;
;
;
;
;
;
;
;
;
```

**Comment by viciious** [comment]
> @viciious because there are empty defines, when they get expanded we end up having extra semicolons. For example, on systems other than Windows this:
> 
> ```
> QGL_WGL( PROC, wglGetProcAddress, ( LPCSTR ) );
> QGL_WGL( int, wglChoosePixelFormat, ( HDC, CONST PIXELFORMATDESCRIPTOR * ) );
> QGL_WGL( int, wglDescribePixelFormat, ( HDC, int, UINT, LPPIXELFORMATDESCRIPTOR ) );
> QGL_WGL( BOOL, wglSetPixelFormat, ( HDC, int, CONST PIXELFORMATDESCRIPTOR * ) );
> QGL_WGL( BOOL, wglSwapBuffers, ( HDC ) );
> QGL_WGL( HGLRC, wglCreateContext, ( HDC ) );
> QGL_WGL( BOOL, wglDeleteContext, ( HGLRC ) );
> QGL_WGL( BOOL, wglMakeCurrent, ( HDC, HGLRC ) );
> QGL_WGL( BOOL, wglShareLists, ( HGLRC, HGLRC ) );
> ```
> 
> Gets expanded to:
> 
> ```
> ;
> ;
> ;
> ;
> ;
> ;
> ;
> ;
> ;
> ```

Right

**Comment by ghost** [comment]
@viciious btw, I'd like to make some research. Can you please explain the structure of qfusion's code so that's I can do it faster?

**Comment by viciious** [comment]
https://github.com/Qfusion/qfusion/wiki/Engine-Quick-Start-Guide

**Comment by ghost** [comment]
Ok, thanks, that wiki is incomplete but it'll help me to move further.

**Comment by ghost** [comment]
Also I am afraid qfusion won't run at Wayland while it uses GLX instead of EGL.

**Comment by viciious** [comment]
> Also I am afraid qfusion won't run at Wayland while it uses GLX instead of EGL.

The native X/GLX backend will eventually be phased out in favor of SDL anyway.

**Comment by viciious** [comment]
Thanks!

**Pull request merged**